### PR TITLE
[5.6] Add URL helpers for signed routes

### PR DESCRIPTION
**Pull request by gauravmak**
Users can use the helper functions `signedRoute()` and `temporarySignedRoute()` for generating signed URLs.

### CODE REVIEW

File: src/Illuminate/Foundation/helpers.php [hunk 1]
```diff
@@ -811,6 +811,36 @@ function route($name, $parameters = [], $absolute = true)
     }
 }
 
+if (! function_exists('signedRoute')) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/helpers.php]: Please use snake_case for all global functions.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by sisve** [comment]
I don't know why, but I have gotten the impression that we're no longer adding additional global helpers and prefer to use facades instead. Anyone else gotten this impression too?

Using `URL::signedRoute(...)` isn't _that_ long.

**Comment by gauravmak** [comment]
@GrahamCampbell Done thanks.

@sisve Personally I try to avoid using Facades in view files. `url()->signedRoute()` is an easy option that is available though.

**Comment by taylorotwell** [comment]
No plans to add anymore helpers when possible.

**Comment by devcircus** [comment]
#23645

**Pull request closed without merging**